internal/monitors/collectd/kong/kong.go

```diff
@@ -0,0 +1,164 @@
+// +build !windows
+
+package kong
+
+//go:generate collectd-template-to-go kong.tmpl
+
+import (
+	"github.com/signalfx/signalfx-agent/internal/core/config"
+	"github.com/signalfx/signalfx-agent/internal/monitors"
+	"github.com/signalfx/signalfx-agent/internal/monitors/collectd"
+)
+
+const monitorType = "collectd/kong"
+
+// MONITOR(collectd/kong): Monitors a Kong instance using [collectd-kong](https://github.com/signalfx/collectd-kong).
+//
+// See the [integration documentation](https://github.com/signalfx/integrations/tree/master/collectd-kong)
+// for more information.
+//
+// Sample YAML configuration:
+//
+// ```yaml
+// monitors:
+//   - type: collectd/kong
+//     host: 127.0.0.1
+//     port: 8001
+//     metrics:
+//       - metric: request_latency
+//         report: true
+//       - metric: connections_accepted
+//         report: false
+// ```
+//
+// Sample YAML configuration with custom /signalfx route and white and blacklists
+//
+// ```yaml
+// monitors:
+//   - type: collectd/kong
+//     host: 127.0.0.1
+//     port: 8443
+//     url: https://127.0.0.1:8443/routed_signalfx
+//     authHeader:
+//       header: Authorization
+//       value: HeaderValue
+//     metrics:
+//       - metric: request_latency
+//         report: true
+//     reportStatusCodeGroups: true
+//     statusCodes:
+//       - 202
+//       - 403
+//       - 405
+//       - 419
+//       - "5*"
+//     serviceNamesBlacklist:
+//       - "*SomeService*"
+// ```
+//
+
+func init() {
+	monitors.Register(monitorType, func() interface{} {
+		return &Monitor{
+			*collectd.NewMonitorCore(CollectdTemplate),
+		}
+	}, &Config{})
+}
+
+// Header defines name/value pair for AuthHeader option
+type Header struct {
+    // Name of header to include with GET
```

Comment:
Run this through `go fmt`.  The tabs/spaces are off.

Reply:
Golang uses tabs for the base indent levels and spaces for alignment.

Reply:
updated.